pkg/types/conf_test.go

```diff
@@ -639,6 +640,39 @@ var _ = Describe("config operations", func() {
 		Expect(rt.CapabilityArgs["portMappings"]).To(Equal(rc.PortMaps))
 	})
 
+	It("creates a valid CNI runtime config with K8s args passed via CNI_ARGS environment variable", func() {
+		args := &skel.CmdArgs{
+			ContainerID: "123456789",
+			Netns:       testNS.Path(),
+			IfName:      "eth0",
+			StdinData: []byte(`{
+    "name": "node-cni-network",
+    "type": "multus",
+    "defaultnetworkfile": "/tmp/foo.multus.conf",
+    "defaultnetworkwaitseconds": 3,
+    "delegates": [{
+        "name": "weave1",
+        "cniVersion": "0.2.0",
+        "type": "weave-net"
+    },{
+        "name": "other1",
+        "cniVersion": "0.2.0",
+        "type": "other-plugin"
+    }]
+}`),
+		}
+
+		os.Setenv("CNI_ARGS", "K8S_POD_NAME=dummy;K8S_POD_NAMESPACE=namespacedummy;K8S_POD_INFRA_CONTAINER_ID=123456789")
+		rt, _ := CreateCNIRuntimeConf(args, nil, "", &RuntimeConfig{}, nil)
+		fmt.Println("rt.ContainerID: ", rt.ContainerID)
+		Expect(rt.ContainerID).To(Equal("123456789"))
+		Expect(rt.NetNS).To(Equal(args.Netns))
+		Expect(rt.IfName).To(Equal(""))
+		Expect(rt.Args[0]).To(Equal([2]string{"K8S_POD_NAME", "dummy"}))
+		Expect(rt.Args[0]).To(Equal([2]string{"K8S_POD_NAMESPACE", "namespacedummy"}))
+		Expect(rt.Args[0]).To(Equal([2]string{"K8S_POD_INFRA_CONTAINER_ID", "123456789"}))
```

Comment:
rt.Args[] is formatted as https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/pkg/types/conf.go#L213-L219 hence the testing code might be following (I also add more ARGS test data in it)

```
                k8sArgs := &K8sArgs{K8S_POD_NAME: "dummy", K8S_POD_NAMESPACE: "namespacedummy", K8S_POD_INFRA_CONTAINER_ID: "123456789"}
                os.Setenv("CNI_ARGS", "K8S_POD_NAME=dummy;K8S_POD_NAMESPACE=namespacedummy;K8S_POD_INFRA_CONTAINER_ID=123456789;DUMMY_ARGS1=test;DUMMY_ARGS2=arg2key=arg2val")
                rt, _ := CreateCNIRuntimeConf(args, k8sArgs, "", &RuntimeConfig{}, nil)
                fmt.Println("rt.ContainerID: ", rt.ContainerID)
                Expect(rt.ContainerID).To(Equal("123456789"))
                Expect(rt.NetNS).To(Equal(args.Netns))
                Expect(rt.IfName).To(Equal(""))
                Expect(rt.Args[0]).To(Equal([2]string{"IgnoreUnknown", "true"}))
                Expect(rt.Args[1]).To(Equal([2]string{"K8S_POD_NAMESPACE", "namespacedummy"}))
                Expect(rt.Args[2]).To(Equal([2]string{"K8S_POD_NAME", "dummy"}))
                Expect(rt.Args[3]).To(Equal([2]string{"K8S_POD_INFRA_CONTAINER_ID", "123456789"}))
                Expect(rt.Args[4]).To(Equal([2]string{"K8S_POD_UID", ""}))
                Expect(rt.Args[5]).To(Equal([2]string{"DUMMY_ARGS1", "test"}))
                Expect(rt.Args[6]).To(Equal([2]string{"DUMMY_ARGS2", "arg2key=arg2val"}))
```

Reply:
Yeah, good catch. I didn't run locally with root privs which would have caught this, I was depending on CI :)